src/algebra/lie_algebra.lean

```diff
@@ -397,6 +416,20 @@ instance lie_subalgebra_lie_algebra (L' : lie_subalgebra R L) :
     @lie_algebra R L' _ (lie_subalgebra_lie_ring _ _ _) :=
 { lie_smul := by { intros, apply set_coe.ext, apply lie_smul } }
 
+local attribute [instance] lie_ring.of_associative_ring
+local attribute [instance] lie_algebra.of_associative_algebra
+
+/-- A subalgebra of an associative algebra is a Lie subalgebra of the associated Lie algebra. -/
+def lie_subalgebra_of_subalgebra (A : Type v) [ring A] [algebra R A]
+  (A' : subalgebra R A) : lie_subalgebra R A :=
+{ lie_mem := λ x y hx hy, by {
+    change ⁅x, y⁆ ∈ A'.carrier, change x ∈ A'.carrier at hx, change y ∈ A'.carrier at hy,
```

Comment:
I don't like the `.carrier`. Shouldn't there be a coercion somewhere?

Reply:
Oh yeah, gross. The coercion exists. I've dropped the `.carrier`s.